docs: [Browser] Define the support browser runtimes #4168 (#5032)
open-telemetry/opentelemetry-js: README.md

> Client instrumentation for the browser is **experimental** and mostly **unspecified**. If you are interested in
> helping out, get in touch with the [Client Instrumentation SIG][client-instrumentation-sig].

Rather that define versions of specific browsers / runtimes, OpenTelemetry sets the minimum supported version based on the
underlying Language features used.

> The current minumum language feature support is set as [ECMAScript 2020](https://www.w3schools.com/jS/js_2020.asp) that are available

Reviewer: Not sure if w3schools is the most _professional_ place to link to, not expecting something as thick as [the spec](https://tc39.es/ecma262/2020/) but maybe [caniuse](https://caniuse.com/sr_es11) since MDN doesn't have an equivalent page?

Author: @dyladan, @pichlermarc Does OpenTelemetry have a specific (official / semi-official) site that we should use for externally referencing language references?

Reviewer 2: @MSNev I'm not aware of any guidelines regarding that - I think we should use a link to the spec as there's no way to misunderstand what we're talking about. If there's a need for additional links we can still add them later on.

in all modern browsers / runtimes.

This means that if you are targeting or your end-users are using a browser / runtime that does not support ES2020, you will need

> to either transpile your code to ensure compatibility or provide any necessary polyfill's for the missing features. Any support

Reviewer: Generally both are needed (transpiling for syntax, polyfills for runtime)

```suggestion
to transpile the code and provide any necessary polyfill's for the missing features to ensure compatibility with your target environments. Any support
```
